browser/src/slideshow/engine/SlideAnimations.ts

```diff
@@ -40,14 +40,16 @@ function getNodeChildren(aNode: ContainerNodeInfo): Array<AnimationNodeInfo> {
 function createAnimationTree(
 	aAnimationRoot: AnimationNodeInfo,
 	aNodeContext: NodeContext,
+	metaSlide: MetaSlide,
 ): BaseNode {
-	return createAnimationNode(aAnimationRoot, null, aNodeContext);
+	return createAnimationNode(aAnimationRoot, null, aNodeContext, metaSlide);
```

Comment:
I think that would be cleaner, to pass the meta slide or just slide info to SlideAnimation ctor and append it to node context which is already passed down along tree nodes.
Anyway nice job.

Reply:
right, done